include/seqan3/std/charconv

```diff
@@ -0,0 +1,1010 @@
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+// -*- C++ -*-
+//===------------------------------ charconv ------------------------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is dual licensed under the MIT and the University of Illinois Open
+// Source Licenses. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+/*!\file
+ * \brief Implements std::from_char.
+ * \author Svenja Mehringer <svenja.mehringer AT fu-berlin.de>
+ */
+
+#pragma once
+
+/*!\defgroup charconv charconv
+ * \ingroup std
+ * \brief The \<charconv\> header from C++20's standard library.
+ */
+
+// -----------------------------------------------------------------------------
+// from_char for integral types
+// -----------------------------------------------------------------------------
+
+#if __has_include(<charconv>)
+#include <charconv>
+#else // implement
+
+#include <cstdlib>
+#include <type_traits>
+#include <limits>
+#include <cstdint>
+#include <cstring>
+#include <cmath>
+#include <cerrno>
+
+#include <seqan3/core/concept/core_language.hpp>
+#include <seqan3/io/stream/parse_condition.hpp>
+#include <seqan3/std/concepts>
+
+namespace std
+{
+//!\brief Result type of std::to_chars
+//!\ingroup charconv
+struct to_chars_result
+{
+    char* ptr;  //!< A pointer to the output string.
+    errc ec;    //!< The error code.
+};
+
+//!\brief Result type of std::from_chars
+//!\ingroup charconv
+struct from_chars_result
+{
+    char const * ptr;  //!< A pointer to the input string.
+    errc ec;          //!< The error code.
+};
+
+//!\cond
+// implementation detail taken from LLVM
+void to_chars(char*, char*, bool, int = 10) = delete;
+void from_chars(char const *, char const *, bool, int = 10) = delete;
+
+namespace itoa
+{
+
+static constexpr uint64_t pow10_64[] =
+{
+    UINT64_C(0),
```

Comment:
what does UINT64_C do?

Reply:
[Appends the correct suffix to a 64-bit unsigned integer literal.](https://jameshfisher.com/2017/01/20/uint64_c_macro.html)